### PR TITLE
[Withdraw] Refactor fee handling

### LINKED ISSUE #1381: ADD: Better Fees Handling
# Desired

A simple rule of thumb to follow. 

This issue relies on implementing this: https://github.com/xchainjs/xchainjs-lib/issues/299 first, which is to change the suggested fees from thirdparty to THORChain gas oracle. 

## FEES
Three fees in THORChain:
1) The on-chain deposit transaction fee (inbound tx)
2) The liquidity fee as a function of slip
3) The on-chain redemption transaction fee (outbound tx)

> For clarity, the two chains will be referred to as the `inboundChain` and the `outboundChain`

### Inbound Fee

Get from `inboundChain.getFee(params, "fast")` -> the fee the user will pay to make the tx

### Liquidity Fee

Do not show to the user, since it is included in the quote

### Outbound Fee

Get from `3 * outboundChain.getFee(params, "fast")` -> the fee that will be deducted from outbound/refund/withdrawal

## THRESHOLDS

If a user swaps/adds liquidity to a chain, two things could happen:
1) swap is successful, the outboundFee will be deducted from the output
2) swap/add fails, the outboundFee will be deducted from the refund

So, do not let the user send an amount less than the higher of the two. 

## Implementation

Firstly, make sure the outboundFee for calculations is computed in the value of the inbound side:
```
outboundFeeInInboundAsset = getValueOfToken1InToken2(outboundFee, outboundAsset , inboundAsset)
```

"Estimated Fee". -> Display this to the user
`1.5 * minInputAmount` -> Do not let the user input less than this (use form validation). Over-estimate by 50% (1.5x) to prevent issues. 

### Swaps 

```
"Estimated Fee" == "inboundFee" + "outboundFee"
minInputAmount = 1.5 * max(inboundFee, outboundFeeInInboundAsset) // swap or refund
```

### Add liquidity 

```
"Estimated Fee" == "inboundFee" + "outboundFee"
minInputAmount = 1.5 * max(inboundFee, outboundFeeInInboundAsset)  // add or refund
```

### Withdraw Liquidity 

```
"Estimated Fee" == "inboundFee" + "outboundFee" // withdrawals
```

### PR DESCRIPTION
**Pull request by veado**
- [x] Rewrite services for withdraw fees
- [x] Update `Withdraw` component
- [x] Quick fixes for symDepositFees$ 

Close #1381

### CODE REVIEW

File: src/renderer/services/chain/fees/withdraw.ts [hunk 1]
```diff
@@ -1,72 +1,77 @@
...
+// State to reload sym deposit fees
+const {
+  get$: reloadWithdrawFees$,
+  get: reloadWithdrawFeeState,
+  set: _reloadSymDepositFees
+} = observableState<O.Option<Asset>>(O.none)
+
+// Triggers reloading of deposit fees
+const reloadWithdrawFees = (asset: Asset) => {
+  console.log('reloadWithdrawFees:', asset)
```
Comment (veado) [inline src/renderer/services/chain/fees/withdraw.ts]: Remove `console.log`

File: src/renderer/components/deposit/withdraw/Withdraw.tsx [hunk 2]
```diff
@@ -113,27 +122,44 @@ export const Withdraw: React.FC<Props> = ({
...
-  const feeLD: FeeLD = useMemo(
-    () => fee$(AssetRuneNative.chain, memo),
+  const prevWithdrawFees = useRef<O.Option<WithdrawFees>>(O.none)
 
-    [fee$, memo]
+  const [withdrawFeesRD] = useObservableState<WithdrawFeesRD>(
+    () =>
+      FP.pipe(
+        fees$(AssetRuneNative),
+        // RxOp.map((v) => v),
```
Comment (thatStrangeGuyThorchain) [inline src/renderer/components/deposit/withdraw/Withdraw.tsx]: nitpick

File: src/renderer/components/deposit/withdraw/Withdraw.tsx [hunk 3]
```diff
@@ -314,6 +336,7 @@ export const Withdraw: React.FC<Props> = ({
         key={'asset amount slider'}
         value={withdrawPercent}
         onChange={setWithdrawPercent}
+        onAfterChange={() => reloadFeesHandler()}
```
Comment (thatStrangeGuyThorchain) [inline src/renderer/components/deposit/withdraw/Withdraw.tsx]: just `reloadFeesHandler`

File: src/renderer/services/chain/fees/withdraw.ts [hunk 4]
```diff
@@ -1,72 +1,76 @@
...
+        oAsset,
+        O.getOrElse(() => initialAsset)
+      )
+
+      return FP.pipe(
+        poolFee$(asset),
+        liveData.map(({ asset: feeAsset, amount: feeAmount }) => ({
+          asset: feeAsset,
+          inFee: feeAmount,
+          outFee: feeAmount.times(3)
```
Comment (thatStrangeGuyThorchain) [inline src/renderer/services/chain/fees/withdraw.ts:76]: pls add a comment about why `3`. just for "future us"

**Review by thatStrangeGuyThorchain — DISMISSED** [review]
NICE 👍

**Pull request merged**